vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -0,0 +1,136 @@
+# (C) Datadog, Inc. 2019-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from pyVmomi import vim
+
+from datadog_checks.base import to_string
+from datadog_checks.vsphere.constants import MOR_TYPE_AS_STRING, REFERENCE_METRIC, SHORT_ROLLUP
+
+
+def format_metric_name(counter):
+    return "{}.{}.{}".format(
+        to_string(counter.groupInfo.key), to_string(counter.nameInfo.key), SHORT_ROLLUP[str(counter.rollupType)],
+    )
+
+
+def match_any_regex(string, regexes):
+    for regex in regexes:
+        match = regex.match(string)
+        if match:
+            return True
+    return False
+
+
+def is_resource_excluded_by_filters(mor, infrastructure_data, resource_filters):
+    resource_type = MOR_TYPE_AS_STRING[type(mor)]
+
+    if not [f for f in resource_filters if f[0] == resource_type]:
+        # No filter for this resource, collect everything
+        return False
+
+    name_filter = resource_filters.get((resource_type, 'name'))
+    inventory_path_filter = resource_filters.get((resource_type, 'inventory_path'))
+    hostname_filter = resource_filters.get((resource_type, 'hostname'))
+    guest_hostname_filter = resource_filters.get((resource_type, 'guest_hostname'))
+
+    if name_filter:
+        mor_name = infrastructure_data.get(mor).get("name", "")
+        if match_any_regex(mor_name, name_filter):
+            return False
+    if inventory_path_filter:
+        path = make_inventory_path(mor, infrastructure_data)
+        if match_any_regex(path, inventory_path_filter):
+            return False
+
+    if hostname_filter and isinstance(mor, vim.VirtualMachine):
+        host = infrastructure_data.get(mor).get("runtime.host")
+        hostname = infrastructure_data.get(host, {}).get("name", "")
+        if match_any_regex(hostname, hostname_filter):
+            return False
+    if guest_hostname_filter and isinstance(mor, vim.VirtualMachine):
+        guest_hostname = infrastructure_data.get(mor, {}).get("guest.hostName", "")
+        if match_any_regex(guest_hostname, guest_hostname_filter):
+            return False
+
+    return True
+
+
+def is_metric_excluded_by_filters(metric_name, mor_type, metric_filters):
+    if metric_name.startswith(REFERENCE_METRIC):
```

Comment:
1. Could you add a test for this case?

2. Could you add to the comment why we need at least one metric for reference?

Reply:
Done